Consul.Test/BaseFixture.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Consul.Test
+{
+    public class BaseFixture : IAsyncLifetime
+    {
+        protected ConsulClient _client;
+        private static bool _ready;
+
+        public BaseFixture()
+        {
+            _client = new ConsulClient(c =>
+            {
+                c.Token = TestHelper.MasterToken;
+                c.Address = TestHelper.HttpUri;
+            });
+        }
+
+        public Task DisposeAsync()
+        {
+            _client.Dispose();
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// With parallel testing on the CI, sometimes the tests start before the consul server finished properly initializing
+        /// This aims to workaround it in a not so elegant way. https://github.com/hashicorp/consul/issues/819
+        /// </summary>
+        /// <returns></returns>
+        public Task InitializeAsync()
```

Comment:
I think it is better to avoid non-async methods returning `Task` as they are more difficult to reason about:
```suggestion
        public async Task InitializeAsync()
```